localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/integration_response.py

```diff
@@ -132,3 +179,59 @@ def select_integration_response(
                     selected_response["statusCode"],
                 )
         return selected_response
+
+    @staticmethod
+    def get_response_template(
+        integration_response: IntegrationResponse, request: InvocationRequest
+    ) -> str:
+        """The Response Template is selected from the response templates.
+        If there are no templates defined, the body will pass through.
+        Apigateway looks at the integration request `Accept` header and defaults to `application/json`.
+        If no template is matched, Apigateway will use the "first" existing template and use it as default.
+        https://docs.aws.amazon.com/apigateway/latest/developerguide/request-response-data-mappings.html#transforming-request-response-body
+        """
+        if not (response_templates := integration_response["responseTemplates"]):
+            return ""
+
+        # The invocation request header is used to find the right response templated
+        accept = request["headers"].get("accept")
```

Comment:
I also found this in the documentation: 
> When a request contains multiple media types in its Accept header, API Gateway only honors the first Accept media type.

I guess it depends it's a proper header list, in that case `get` might get the first, or if we need to parse the value and split on `,`. Maybe we could write a quick test for it?

Reply:
There will be a follow up pr with it that will include a fix for the headers